blog/ca/3_web-scraping-de-feines-a-linkedIn-utilitzant-puppeteer-i-rxjs.mdx

```diff
@@ -0,0 +1,502 @@
+            }
+        }
+        return results;
+    }, {stacks})) as Observable<JobInterface[]>)
+}
+
+```
+
+El codi proporcionat extreu efectivament tota la informació de treball disponible de la pàgina. Encara que el codi és molt estètic, aconsegueix la feina, que és típic per a codi de "scraping" web.
+
+> En un context de programació estàndard, generalment és aconsellable descompondre el codi en funcions més petites i aïllades per millorar la llegibilitat i la mantenibilitat. No obstant això, quan es tracta de codi executat dins de `page.evaluate` en Puppeteer, estem una mica limitats perquè aquest codi s'executa en la instància de Puppeteer (Chrome), no en el nostre entorn Node.js. Per tant, tot el codi ha de ser autocontingut dins de la crida de `page.evaluate`. L'única excepció aquí són les variables (com `stacks` en el nostre cas), que poden passar-se com a arguments a `page.evaluate`, sempre que no continguin funcions o objectes complexos que no es puguin serialitzar.
+
+In this case, the only challenging part to scrape is the salary information, as it involves converting a text format like '$65,000.00 - $90,000.00' into separate salaryMin and salaryMax values.
+Additionally, we've encapsulated the entire code within a try/catch block to gracefully handle errors. While we currently log errors to the console, it's advisable to consider implementing a mechanism to store these error logs on disk. This practice becomes particularly important because web pages often undergo changes, necessitating frequent updates to the HTML parsing code.
+
+En aquest cas, l'únic component desafiant "per escrapejar"(to scrape) és la informació del salari, ja que implica convertir un format de text com '$65,000.00 - $90,000.00' en valors de salari mínim i màxim separats.
+A més, hem encapsulat tot el codi dins d'un bloc try/catch per gestionar els errors de manera elegant. Encara que actualment registrem els errors a la consola, és aconsellable considerar la implementació d'un mecanisme per emmagatzemar aquests registres d'error en disc. Aquesta pràctica es torna particularment important perquè les pàgines web sovint experimenten canvis, cosa que necessita actualitzacions freqüents al codi de l'anàlisi HTML.
+
+
+Finalment, és important notar que sempre utilitzem els operadors `defer` i `fromPromise` per convertir Promeses en Observables:
+
+```typescript
+defer(() => fromPromise(myPromise()))
+```
+This approach is a recommended best practice that works reliably in all scenarios. Promises are eager, whereas Observables are lazy and only initiate when someone subscribes to them. The defer operator allows us to make a Promise lazy.
+
+Aquest enfocament és una millor pràctica recomanada que funciona de manera fiable en tots els escenaris. Les Promeses són "eager", mentre que els Observables són "lazy" i només s'inicien quan algú s'hi subscriu. L'operador `defer` ens permet convertir a "lazy" una Promesa.
```

Comment:
"best practice" can be translated as "bones pràctiques"

Reply:
Can you also add a link to a page that explains the difference between a promise and an observable? I think it can be handy

Reply:
Pot estar bé si, si tens algun enllaç bo pots passar-me'l, sinó en busco algun.

Reply:
Ho he substituit per:

Aquesta estratègia és la més recomanada ja que funciona de manera fiable en tots els escenaris